### PR TITLE
[DOCFIX] Update POSIX API doc

### PR DESCRIPTION
**Pull request by apc999**
Improve and POSIX API docs, especially simplifying the setting due to https://github.com/Alluxio/alluxio/pull/12857/

### CODE REVIEW

File: docs/en/api/POSIX-API.md [hunk 1]
```diff
@@ -42,13 +42,12 @@ The Alluxio POSIX API has two implementations for users to choose from:
 Alluxio's default Fuse implementation that uses [JNR-Fuse](https://github.com/SerCeMan/jnr-fuse) for FUSE on Java.
 JNR-Fuse targets for low concurrency scenarios and has some known limitations in performance.
 * Alluxio JNI-Fuse (Experimental)
-Alluxio JNI-Fuse is a new in-house FUSE implementation based on JNI (Java Native Interface) which targets for challenging AI scenarios, initiated by Nanjing University and Alibaba.
-JNI-Fuse is still experimental but it is under active development by Alluxio community to solve the compatibility issues and improve stability and performance.
+This is a new in-house implementation based on JNI (Java Native Interface) which targets for more performance-sensitve applications (like model training workloads) and initiated by researchers from Nanjing University and engineers from Alibaba Inc.
```
Comment (LuQQiu) [inline docs/en/api/POSIX-API.md:45]: targets more performance-sensitive applications

File: docs/en/api/POSIX-API.md [hunk 2]
```diff
@@ -42,13 +42,12 @@ The Alluxio POSIX API has two implementations for users to choose from:
...
 
-To choose between the default JNR-Fuse and JNI-Fuse, here are some aspects to consider:
+Here is a guideline To choose between the default JNR-Fuse and experimental JNI-Fuse:
 
-* Ease of usage: JNR-Fuse is enabled by default while JNI-Fuse has more dependencies.
-* Maintenance: JNI-Fuse is experimental but is the direction to go. Alluxio will focus more on developing JNI-Fuse.
+* Workloads: If your data access pattern is highly concurrent (e.g., deep learning training), JNI-Fuse is better and more stable for this workloads.
```
Comment (LuQQiu) [inline docs/en/api/POSIX-API.md:49]: these workloads.

**Review by LuQQiu — APPROVED** [review]
Left two minor comments, otherwise LGTM

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**